### DISCUSSION TITLE
no design documentation , only buzz words

### DISCUSSION THREAD

**Discussion by raphaelauv** [crate/crate#12945] [Q&A]
hey , when I visit the homepage of https://crate.io/ or https://github.com/crate/crate

it's a list of commercial buzzwords and not a single factual and pragmatic explanation of the technical design of crate

so I still don't know after 2 minutes looking around what is really design for crate , is it a distributed acid column oriented database for OLTP ?

**Comment by proddata** [comment]
Hi @raphaelauv 

Thanks for the feedback. May I ask you to provide example, what you might want to see?

> so I still don't know after 2 minutes looking around what is really design for crate , is it a distributed acid column oriented database for OLTP ?

CrateDB definitely more a DB for OLAP use cases, as we don't support transactions. However OLAP is often brought into connection with specific database schemas / cube models, which many modern "analytics" databases don't follow / support.

_In short:_
CrateDB is an eventual consistent distributed SQL databases for real-time analytics (i.e. ad-hoc queries and aggregations) ​across large datasets of (semi)-structured data built on top of Apache Lucene. While using columnar structures, CrateDB is not a pure columnar store, but rather combines document and columnar data storage with distributed inverted indexes (with full-text parsers / analysers) and BKD-trees.
